scylla-cdc-base/pom.xml

```diff
@@ -9,4 +9,16 @@
     <name>Java Library for Scylla CDC - base</name>
     <description>A library for reading the Scylla CDC log.</description>
     <packaging>jar</packaging>
+
+    <properties>
+        <guava.version>19.0</guava.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${guava.version}</version>
+        </dependency>
+    </dependencies>
```

Comment:
Wouldn't we want to use shaded guava in all modules. What good does it make to shade guava in one module when we use it in another?

Reply:
In scylla-cdc-base and scylla-cdc-lib, Guava is pretty much used only for `Preconditions` (which I assume is stable between Guava releases). The original reasoning behind shading was that the driver-specific code used some Future-related methods from Guava, incompatible with newer Guava versions (therefore in current version of this PR, only Guava in scylla-cdc-driver3 is shaded).

What would be a proper approach to shade Guava in all these three modules (scylla-cdc-base, scylla-cdc-lib, scylla-cdc-driver3)? 

- Shade it in top-level one (scylla-cdc-base) and then in other modules carefully import shaded classes from scylla-cdc-base (no shading in those other modules)?
- Do the shading separately in each module?

Reply:
Yes. Shading it in a module used by everything else would be best.